### PR TITLE
[gcp-deployer] Add Gapi manager class, more typings and fixes

### PR DESCRIPTION
**Pull request by yebrahim**
This PR mainly:
- Adds a new separate `Gapi` class to manage the gapi runtime and add more friendly wrappers.
- Loads gapi by listening on a global promise object.
- Adds some typings and fixes small bugs that popped up as a result.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1054)

### CODE REVIEW

**Comment by yebrahim** [comment]
/assign jlewi
/assign kunmingg

Let me know if I should assign someone else.

File: components/gcp-click-to-deploy/src/Gapi.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  public static deploymentmanager = class {
+
+    public static async insert(project: string, resource: {}) {
+      await this._load();
+      return this._deploymentManager.insert({ project, resource } as any).then(r => r.body);
+    }
+
+    public static async list() {
+      await this._load();
+      return this._deploymentManager.list({ project: 'yelsayed-project1' }).then(r => r.body);
```
Comment (jlewi) [inline components/gcp-click-to-deploy/src/Gapi.ts]: Did you mean to leave your project here?

Comment (yebrahim) [reply components/gcp-click-to-deploy/src/Gapi.ts]: Just leaving my mark on the project! :)
Thanks for the catch, removed.

**Comment by yebrahim** [comment]
@jlewi how can I see the run errors?

**Pull request merged**